elastic/kibana
# [Security Solution] Timeline : Disabling Timeline ESQL feature flag should disable ESQL Tab.

## Summary

handles https://github.com/elastic/kibana/issues/182798

Recently there was PR : https://github.com/elastic/kibana/pull/181616 which does not disable ES|QL tab in timeline even if feature flag is disabled when : 
- User has already created a ESQL Query in timeline and saved the timeline.

This PR makes sure when below feature flag exists, then `ES|QL` tab will be definitely disabled even when user has a saved timeline with ES|QL Query in it. 

```yaml
xpack.securitySolution.enableExperimental:
  - timelineEsqlTabDisabled

```

## Desk Testing Guidelines

1. Remove above Feature Flag
2. Go to Timeline and Create a Timeline with ESQL Query
3. Save the timeline.
4. Go To advanced Settings and search for `esql` and disable the `enableESQL` setting.
5. Go back to the timeline saved in step 3. 
6. ✅ Assert that the `ES|QL` tab is still there.
7. ✅ Assert that the `ES|QL` tab is NOT there in a new timeline.
8. Add above feature flag - which disables the esql Tab.
9. Go back to the timeline saved in step 3.
10. ✅ Assert that the `ES|QL` tab is no longer there.
11. ✅ Assert that the `ES|QL` tab is NOT there in a new timeline.


### Checklist
